File: src/Core/FinancialEvents/Schemas/LotMatchingConfiguration.py
from typing import Callable

from arrow import Arrow

import Core.FinancialEvents.Schemas.Grouping as pgf
from Core.LotMatching.Contracts.LotMatchingMethod import LotMatchingMethod


class LotMatchingConfiguration:
    ForStocks: Callable[[pgf.FinancialGrouping], LotMatchingMethod]
    ForDerivatives: Callable[[pgf.FinancialGrouping], LotMatchingMethod]

    fromDate: Arrow
    toDate: Arrow

    def __init__(
        self,
        fromDate: Arrow,
        toDate: Arrow,
        forStocks: Callable[[pgf.FinancialGrouping], LotMatchingMethod],
        forDerivatives: Callable[[pgf.FinancialGrouping], LotMatchingMethod],
    ) -> None:
        self.ForStocks = forStocks
        self.ForDerivatives = forDerivatives
        self.fromDate = fromDate
        self.toDate = toDate

File: lib/src/Core/Schemas/LotMatchingConfiguration.ts
import type { FinancialGrouping } from "@brrr/Core/Schemas/Grouping";
import type { ValidDateTime } from "@brrr/Utils/DateTime";

// Forward declaration - LotMatchingMethod is defined in lotMatching
export interface LotMatchingMethodContract {
	performMatching(events: unknown[]): unknown[];
}

export type LotMatchingConfiguration = {
	fromDate: ValidDateTime;
	toDate: ValidDateTime;
	forStocks: (grouping: FinancialGrouping) => LotMatchingMethodContract;
	forDerivatives: (grouping: FinancialGrouping) => LotMatchingMethodContract;
};
